opendc-model-odc/core/src/main/kotlin/com/atlarge/opendc/model/odc/platform/scheduler/stages/machine/Selection.kt

```diff
@@ -92,3 +92,29 @@ class RandomMachineSelectionPolicy(private val random: Random = Random()) : Mach
         else
             null
 }
+
+/**
+ * Round robin (RR) scheduling.
+ *
+ * https://en.wikipedia.org/wiki/Round-robin_scheduling
+ */
+class RrMachineSelectionPolicy(private var next: Int = 0) : MachineSelectionPolicy {
+    override suspend fun select(machines: List<Machine>, task: Task): Machine? =
+        context<StageScheduler.State, OdcModel>().run {
+            model.run {
+                // Try to find the machine with id `next`, if that machine can't
+                // be found try `next + 1`, etc.
+                val ids = machines.map { machine -> machine.id }
+                val max_id: Int = ids.max() ?: Int.MAX_VALUE
+                while (ids.size > 0) {
+                    val index = ids.indexOf(next)
```

Comment:
Wouldn't it be smarter to do a Round Robin over the indices of the `machines` list instead, to prevent many iterations here for sparse identifiers. Alternatively, you could sort machines by identifier and do the Round Robin then.

Reply:
I don't think doing it based on indices would result in a proper round robin algorithm.

Say we have machines `[1, 3, 4]` and `next = 1`. The next call to `select` will result in machine 3 (with index 1) being selected, ending with `next = 2`. Before the next call to select machines 2 and 3 are already done. That means for the next call to select we have `[1, 2, 3, 4]` machines and `next = 2`. Which means that machine 3 will be selected again, twice in a row.

Reply:
I've pushed a change that improve the performance a bit. But in our experiments Round Robin was by no means the slowest.

Reply:
Okay, let's keep it like this then!